src/main/java/org/assertj/core/api/AbstractAssert.java

```diff
@@ -830,4 +834,23 @@ private void propagateAssertionInfoFrom(AbstractAssert<?, ?> assertInstance) {
     return usingRecursiveComparison(new RecursiveComparisonConfiguration());
   }
 
+  @SuppressWarnings("unchecked")
+  protected <ASSERT extends AbstractAssert<?, ?>> ASSERT extracting(String propertyOrField,
+                                                                    AssertFactory<Object, ASSERT> assertFactory) {
+    requireNonNull(assertFactory, shouldNotBeNull("assertFactory").create());
+    Object value = byName(propertyOrField).apply(actual);
+    String extractedPropertyOrFieldDescription = extractedDescriptionOf(propertyOrField);
+    String description = mostRelevantDescription(info.description(), extractedPropertyOrFieldDescription);
+    return (ASSERT) assertFactory.createAssert(value).withAssertionState(myself).as(description);
+  }
+
+  @SuppressWarnings("unchecked")
+  protected <T, ASSERT extends AbstractAssert<?, T>> ASSERT extracting(Function<? super ACTUAL, T> extractor,
```

Comment:
To use this to extract to an ObjectAssert, I assume I would call 

```java
extracting(actual.someMethod(), Assertions::assertThatObject);
```

?

Reply:
I would say `extracting(actual::someMethod, Assertions:assertThat)` if you want to have the corresponding typed assertion according to the extractor result, e.g.:
```java
extracting(actual::toString, Assertions:assertThat) // returns AbstractStringAssert<?>
```
or use `assertThatObject` if you really want to have an `ObjectAssert`:
```java
extracting(actual::toString, Assertions:assertThatObject) // returns ObjectAssert<String>
```